tokio-console/src/conn.rs

```diff
@@ -112,6 +114,48 @@ impl Connection {
                     Some("file") => {
                         return Err("unix domain sockets are not supported on this platform".into());
                     }
+                    #[cfg(feature = "vsock")]
+                    Some("vsock") => {
+                        if !matches!(self.target.host(), None | Some("localhost") | Some("any")) {
+                            return Err("cannot connect to non-localhost vsock".into());
+                        }
+
+                        // Parse URI path in the format vsock://<cid>:<port>
+                        let uri_path = self.target.path();
+                        let parts: Vec<&str> =
+                            uri_path.trim_start_matches('/').split(':').collect();
+                        if parts.len() != 2 {
+                            return Err(format!(
+                                "invalid vsock URI format, expected vsock://<cid>:<port>, got {}",
+                                self.target
+                            )
+                            .into());
+                        }
+
+                        let cid = parts[0]
+                            .parse::<u32>()
+                            .map_err(|_| format!("invalid CID: {}", parts[0]))?;
+                        let port = parts[1]
+                            .parse::<u32>()
+                            .map_err(|_| format!("invalid port: {}", parts[1]))?;
+
+                        let vsock_addr = tokio_vsock::VsockAddr::new(cid, port);
+
+                        // Dummy endpoint is ignored by the connector
+                        let endpoint = Endpoint::from_static("http://localhost");
+                        endpoint
+                            .connect_with_connector(tower::service_fn(move |_| {
+                                VsockStream::connect(vsock_addr).map_ok(TokioIo::new)
+                            }))
+                            .await?
+                    }
+                    #[cfg(not(feature = "vsock"))]
+                    Some("vsock") => {
+                        return Err(
+                            "vsock is not supported on this platform or feature is not enabled"
```

Comment:
We can probably limit this to saying the feature isn't enabled, since that's why we've hit this code path.